adr/0007-placement-support.md

```diff
@@ -0,0 +1,121 @@
+* straightforward transition process once infrastructure in place
+
+### Negative Consequences <!-- optional -->
+
+* Expansion of semantics of existing concepts may be harder to document, understand and explain than dedicated new concepts
+
+## Design Detail
+
+### New Link Semantics
+
+Currently links are used extensively in product solutions. A link goes from an occurrence in one solution to an occurrence in another (possibly same) solution. This option builds on existing link processing as follows;
+
+1. A link targets an occurrence within a solution.
+1. A link uses attributes to identify a solution and the specific occurrence within that solution.
+1. Occurrences within a solution are identified by a standard set of link attributes
+     - `Tier`
+     - `Component`
+     - `Subcomponent`
+     - `Instance`
+     - `Version`
+
+   Thus these attributes must be present on a link, or be inherited from the occurrence on which the link is configured.
+1. The `Subcomponent` attribute replaces use of type specific attributes such as `Port`.
+1. An optional `Type` attribute is added to ensure links to same named subcomponents can be resolved. Its value is the component type of the link target.
+1. Each solution has a scope and an identifier within the scope.
+1. A link must have a `Scope` attribute representing the solution scope. The default, representing existing usage, is the `segment` scope.
+1. The scope value determines what other ordered attributes of the link are used to identify the specific solution required.
+1. The attributes to identify the particular solution must either be present on the link, or inherited from the occurrence on which
+the link is configured.
+1. For the `segment` scope, the ordered attributes used to identify a specific solution are `Tenant`, `Product`, `Environment` and `Segment`.
+1. Link indirection is supported by an optional `LinkRef` attribute within a link object. If present, no other link attributes should be explicitly provided.
+1. The value of a LinkRef attribute is used as the attribute name in a `LinkRefs` configuration object, with the attribute value
+being the desired link definition.
+1. LinkRef attributes provide a convenient mechanism to centralise qualification of links shared across multiple components, as is commonly the case with placements.
+
+### New Components
+
+This option also introduces two new components - `Subscription` and `HostingPlatform`.
+
+A `Subscription` component represents provider specific mechanism for purchasing hosting capability, such as an `account` with AWS or a `subscription` with Azure.
+
+Subscription components will typically be used with a `tenant` scoped solution. Instances of a tenant scoped solution will be identified by the `Tenant` link attribute.
+
+A `HostingPlatform` component represents the place within a subscription where resources can physically be hosted. An example for AWS or Azure would be a region within a `Subscription`.
```

Comment:
I think this could use some clarification here. I understand the concept of `HostingPlatform` broadly, but how do `HostingPlatform`-components differentiate from `Region`?

Does a `HostingPlatform` set a limitation - no component can deploy outside of these locations? Or does it represent the capability to deploy to that location?

Reply:
The capability to deploy to that location. I'll add some extra text. I think the idea may evolve as we implement it.